lib/api/v3/activities/activity_representer.rb

```diff
@@ -27,36 +27,16 @@
 # See doc/COPYRIGHT.rdoc for more details.
 #++
 
-require 'roar/decorator'
-require 'roar/json/hal'
-
 API::V3::Utilities::DateTimeFormatter
 
 module API
   module V3
     module Activities
-      class ActivityRepresenter < Roar::Decorator
-        include Roar::JSON::HAL
-        include Roar::Hypermedia
-        include API::V3::Utilities::PathHelper
+      class ActivityRepresenter < ::API::Decorators::Single
         include API::V3::Utilities
 
-        self.as_strategy = API::Utilities::CamelCasingStrategy.new
-
-        def initialize(model, options = {})
-          @current_user = options[:current_user]
-
-          super(model)
-        end
-
-        property :_type, exec_context: :decorator
-
-        link :self do
-          {
-            href: api_v3_paths.activity(represented.id),
-            title: "#{represented.id}"
-          }
-        end
+        self_link path: 'activity',
+                  title_getter: -> (*) { represented.id }
```

Comment:
It is valid and intentionally possible to return `nil` for the title. This means "no title".

You found a case where that makes sense.

P.S.: Just noticed that previously the id was used as title... I don't think that makes sense ^^